backtracking/n_queens_math.py

```diff
@@ -0,0 +1,140 @@
+"""
+ Problem:
+
+ The n queens problem is of placing N queens on a N * N
+ chess board such that no queen can attack any other queens placed
+ on that chess board.
+ This means that one queen cannot have any other queen on its horizontal, vertical and
+ diagonal lines.
+
+ Solution:
+
+ To solve this problem we will use simple math. First we know the queen can move in all
+ the possible ways, we can simplify it in this: vertical, horizontal, diagonal left and
+ diagonal right.
+
+ We can visualize it like this: left diagonal = \ and right diagonal = /
+
+ On a chessboard vertical movement could be the rows and horizontal movement could be the columns.
```

Comment:
```suggestion
 On a chessboard vertical movement could be the rows and horizontal movement could be
 the columns.
```
Lines must be 88 characters max.